pkg/addons2/adminconsole/install.go

```diff
@@ -0,0 +1,194 @@
+package adminconsole
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/embedded-cluster/pkg/addons/registry"
+	"github.com/replicatedhq/embedded-cluster/pkg/helm"
+	"github.com/replicatedhq/embedded-cluster/pkg/kotscli"
+	"github.com/replicatedhq/embedded-cluster/pkg/spinner"
+	"github.com/replicatedhq/embedded-cluster/pkg/versions"
+	"golang.org/x/crypto/bcrypt"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (a *AdminConsole) Install(ctx context.Context, kcli client.Client, writer *spinner.MessageWriter) error {
+	// some resources are not part of the helm chart and need to be created before the chart is installed
+
+	if err := createNamespace(ctx, kcli, namespace); err != nil {
+		return errors.Wrap(err, "create namespace")
+	}
+
+	if err := createKotsPasswordSecret(ctx, kcli, namespace, a.Password); err != nil {
+		return errors.Wrap(err, "create kots password secret")
+	}
+
+	if err := createKotsCAConfigmap(ctx, kcli, namespace, a.PrivateCAs); err != nil {
+		return errors.Wrap(err, "create kots CA configmap")
+	}
+
+	if a.AirgapBundle != "" {
+		err := createRegistrySecret(ctx, kcli, namespace)
+		if err != nil {
+			return errors.Wrap(err, "create registry secret")
+		}
+	}
+
+	// install the helm chart
+
+	helm, err := helm.NewHelm(helm.HelmOptions{
+		K0sVersion: versions.K0sVersion,
+	})
+	if err != nil {
+		return errors.Wrap(err, "create helm client")
+	}
+
+	_, err = helm.Install(ctx, releaseName, Metadata.Location, Metadata.Version, helmValues, namespace)
+	if err != nil {
+		return errors.Wrap(err, "install admin console")
+	}
+
+	// install the application
```

Comment:
do we need to do anything here to make sure it's ready / running before the next step?

Reply:
the `--wait` flag is set to true in the helm install, which waits for things to be ready before moving on